src/data/complex/is_R_or_C.lean

```diff
@@ -362,7 +362,7 @@ by simp only [←sqrt_norm_sq_eq_norm, norm_sq_conj]
 /-! ### Cast lemmas -/
 
 @[simp, norm_cast, priority 900] theorem of_real_nat_cast (n : ℕ) : ((n : ℝ) : K) = n :=
-of_real_hom.map_nat_cast n
+show (algebra_map ℝ K) n = n, from map_nat_cast of_real_hom n
```

Comment:
Can we now revert this to:
```suggestion
map_nat_cast of_real_hom n
```

Reply:
this was the only one that didn't typecheck :/ I'm not sure why, I was thinking of asking Mario on zulip as he was just talking about non-transitive defeq earlier today

Reply:
Aha, this looks like it's same issue as in #11238: sometimes Lean can't match a concrete instance such as `ring_hom.has_coe_to_fun` with a general instance `ring_hom_class.to_has_coe_to_fun` if the types don't quite give enough information yet to infer a `ring_hom_class` instance. I can't find an easy tweak in the defeq check to make this work, so the `show` will have to stay.